File: pages/SecureAreaPage.ts
import { Page, Locator } from '@playwright/test';
import { BasePage } from './BasePage';

/**
 * SecureAreaPage — Page Object for /secure
 * Reached after a successful login.
 */
export class SecureAreaPage extends BasePage {
  readonly heading: Locator;
  readonly logoutButton: Locator;
  readonly flashMessage: Locator;

  constructor(page: Page) {
    super(page);
    this.heading      = page.getByRole('heading', { name: 'Secure Area' });
    this.logoutButton = page.getByRole('link', { name: 'Logout' });
    this.flashMessage = page.locator('#flash');
  }

  async getFlashMessage(): Promise<string> {
    await this.flashMessage.waitFor({ state: 'visible' });
    return (await this.flashMessage.innerText()).trim();
  }

  async logout(): Promise<void> {
    await this.logoutButton.click();
  }
}

File: selenium-java-pom/src/main/java/com/ailearning/seleniumpom/pages/SecureAreaPage.java
package com.ailearning.seleniumpom.pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

public class SecureAreaPage extends BasePage {
    private static final By HEADING = By.cssSelector("h2");
    private static final By LOGOUT_BUTTON = By.cssSelector("a.button.secondary.radius");
    private static final By FLASH_MESSAGE = By.id("flash");

    public SecureAreaPage(WebDriver driver) {
        super(driver);
    }

    public boolean isLoaded() {
        return waitForVisible(HEADING).getText().trim().equals("Secure Area");
    }

    public String getFlashMessage() {
        return getNormalizedText(FLASH_MESSAGE);
    }

    public void logout() {
        waitForVisible(LOGOUT_BUTTON).click();
    }
}
